kafka-plugins-common/src/main/java/io/cdap/plugin/batch/source/KafkaBatchConfig.java

```diff
@@ -190,74 +192,85 @@ public Schema getMessageSchema() {
       // if the field is not the time field and not the key field, it is a message field.
       if (fieldName.equals(keyField)) {
         if (fieldType != Schema.Type.BYTES) {
-          throw new IllegalArgumentException("The key field must be of type bytes or nullable bytes.");
+          collector.addFailure("The key field must be of type bytes or nullable bytes.", null);
         }
         keyFieldExists = true;
       } else if (fieldName.equals(partitionField)) {
         if (fieldType != Schema.Type.INT) {
-          throw new IllegalArgumentException("The partition field must be of type int.");
+          collector.addFailure("The partition field must be of type int.", null);
         }
         partitionFieldExists = true;
       } else if (fieldName.equals(offsetField)) {
         if (fieldType != Schema.Type.LONG) {
-          throw new IllegalArgumentException("The offset field must be of type long.");
+          collector.addFailure("The offset field must be of type long.", null);
         }
         offsetFieldExists = true;
       } else {
         messageFields.add(field);
       }
     }
     if (messageFields.isEmpty()) {
-      throw new IllegalArgumentException(
-        "Schema must contain at least one other field besides the time and key fields.");
+      collector.addFailure(
+        "Schema must contain at least one other field besides the time and key fields.", null);
     }
     if (getKeyField() != null && !keyFieldExists) {
-      throw new IllegalArgumentException(String.format(
-        "keyField '%s' does not exist in the schema. Please add it to the schema.", keyField));
+      collector.addFailure(String.format(
+        "keyField '%s' does not exist in the schema. Please add it to the schema.", keyField), null);
```

Comment:
`.withConfigProperty("keyField")`

should also create a constant  `private static final String KEY_FIELD = "keyField";` and use that everywhere I'm writing "keyField"